### DISCUSSION TITLE
Does milvus 2.3 support HNSW index?

### DISCUSSION THREAD

**Discussion by renzhan** [milvus-io/milvus#30249] [Q&A and General discussion] [answered]
I installed milvus 2.3 standalone on docker, the index type is HNSW, the metric_type is L2, and the gpu is 2080 ti. There are hundreds of thousands of data. After loading the data, I execute a query and find that the gpu has not been occupied. I don’t know why?

**Comment by yhmo** [comment, marked as answer]
To run milvus with gpu:
1. Download the gpu image:  
`docker pull milvusdb/milvus:v2.3.6-gpu`

2. Download the gpu docker-compose.yaml, make sure the prerequisites is met
`wget https://github.com/milvus-io/milvus/releases/download/v2.3.6/milvus-standalone-docker-compose-gpu.yml -O docker-compose.yml`

3. You need to make some changes to the environment variables of the standalone service in the YAML file, read this page for details, the "For GPU users" section:
https://milvus.io/docs/install_standalone-docker.md

3. Use GPU index, only the [GPU_IVF_FLAT](https://milvus.io/docs/index.md#GPU_IVF_FLAT) and [GPU_IVF_PQ](GPU_IVF_PQ) is available for GPU

**Reply by renzhan** [reply]
Thanks, Is there any plan to support HNSW index?

**Reply by yhmo** [reply]
We don't intend to implement HNSW on GPU. There will be another GPU index released in the new version of milvus, maybe in v2.4.
